### PR TITLE
[5.7][silgen] Add support for correctly calling imported throwing methods on imported objc classes that are imported with a non-null NSError.

### PR DESCRIPTION
**Pull request by gottesmm**
This PR contains 3 commits:

1. I added a small helper to the AST to convert a type 'T' in an Optional<T>2. 
2. This is the meat commit. I included it's commit msg below.3. 
3. This just tweaks the test slightly so it only runs where we have objc-interop.

----

[silgen] Add support for correctly calling imported throwing methods on imported objc classes that are imported with a non-null NSError.

This for some time has been crashing in IRGen in non-asserts builds and hitting
assertions in asserts builds. I think we were missing test coverage here and
that is why this basic-ish thing has been broken.

Basically at a high level, the actual expected ABI here is that the NSError is
marked non-nullable which is different from the expected ABI.

rdar://92755102

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Review by jckarter — APPROVED** [review]

**Review by tbkka — APPROVED** [review]
